README.md

```diff
@@ -2,5 +2,435 @@
+
+To prevent it people use [idempotency key](https://stripe.com/docs/api/idempotent_requests) pattern.
+Unit of work allows you to define idempotency key in params, so you can safely make retries.
+```kotlin
+    @Serializable
+    data class Params(
+        val id: String,
+        val currency: String,
+        override val idempotencyKey: IdempotencyKey
+    ) : UowParams<Params> {
+        override fun serialization() = serializer()
+    }
+```
+Idempotency key can be shipped as a standalone artifact outside your service, if you f.e. want to pass it via http request.
+```kotlin
+    implementation("team.razz.eva:eva-idempotency-key:$eva_version")
+```
+
+### Paging
+Out of the box Eva supports paging for your data, when it is not possible to return all results in one request.
+First, you need to add paging module to your dependencies:
+```kotlin
+    implementation("team.razz.eva:eva-paging:$eva_version")
+```
+Second, you need to define your [PagingStrategy](eva-repository/src/main/kotlin/com/razz/eva/repository/PagingStrategy.kt). For now, we support paging by some timestamp only.
+```kotlin
+    object WalletPaging : PagingStrategy<UUID, Wallet.Id, Wallet, Wallet, WalletRecord>(Wallet::class) {
+
+        override fun tableTimestamp() = WALLET.EXPIRE_AT
+    
+        override fun tableId() = WALLET.ID
+    
+        override fun tableOffset(modelOffset: ModelOffset) = UUID.fromString(modelOffset)
+    
+        override fun modelTimestamp(model: Wallet) = model.expireAt
+    
+        override fun modelOffset(model: Wallet) = model.id().stringValue()
+    }
+```
+Now we can implement method in our repository to get pages.
+```kotlin
+    suspend fun wallets(currency: Currency, page: TimestampPage) = findPage(
+        condition = WALLET.CURRENCY.eq(currency.currencyCode),
+        page = page,
+        pagingStrategy = WalletPaging
+    )
+```
+That's all! This method returns object of [PagedList](eva-paging/src/main/kotlin/com/razz/eva/paging/PagedList.kt). It provides a part of your requested results and the next page to query the next part of results.
+
+### Error handling
+One day you are going face a lot of concurrent unit of works.
+It leads to concurrent modification of same models. But our units of work are transactional, so we guarantee consistency of your models.
+In case of concurrent modification unit of work throws [StaleRecordException](eva-persistence/src/main/kotlin/com/razz/eva/persistence/PersistenceException.kt).
+By default, we will do a one retry for such kind of exception, but you can change this strategy
+```kotlin
+val configuration = UnitOfWork.Configuration(
+    retry = StaleRecordFixedRetry(attempts = 3, staleRecordDelay = Duration.ofMillis(100))
+)
+
+class CreateWalletUow(
+    private val queries: WalletQueries,
+    clock: Clock
```

Comment:
TODO: explain why clock is required and how it related to `record_created_at` and `record_updated_at` columns.